build-system/tasks/generate-vendor-jsons.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// TODO: @jonathantyng cleanup #22757
+// this is a temporary gulp task while we migrate amp-analytics to use vendor
+// JSONs instead of JS files
+
+const del = require('del');
+const file = require('gulp-file');
+const gulp = require('gulp');
+const {endBuildStep, toPromise, compileJs} = require('./helpers');
+
+/**
+ * Generate all the vendor config JSONs from their respective JS files
+ * @return {!Promise}
+ */
+function generateVendorJsons() {
+  const startTime = Date.now();
+
+  const srcDir = 'extensions/amp-analytics/0.1';
+  const srcFile = 'vendors.js';
+  const tempPath = 'dist/temp-analytics/';
+  const destPath = 'extensions/amp-analytics/0.1/vendors/';
+  const compileOptions = {
+    browserifyOptions: {
+      // allow gulp to import this file after compile
+      standalone: 'analyticsVendors',
+      // need to stub self since we are not running this in the browser
+      insertGlobalVars: {
```

Comment:
Looks very hacky, but we should get rid of this very soon. So I'm fine with it. 

@erwinmombay @rsimha to make the final call.